pkg/provider/vault/vault.go

```diff
@@ -892,14 +900,18 @@ func (v *client) Validate() (esv1beta1.ValidationResult, error) {
 
 func (v *client) buildMetadataPath(path string) (string, error) {
 	var url string
-	if v.store.Path == nil && !strings.Contains(path, "data") {
-		return "", fmt.Errorf(errPathInvalid)
-	}
-	if v.store.Path == nil {
-		path = strings.Replace(path, "data", "metadata", 1)
-		url = path
+	if v.store.Version == esv1beta1.VaultKVStoreV1 {
+		url = fmt.Sprintf("%s/%s", *v.store.Path, path)
```

Comment:
Why is this change needed? IMO we should probably error out whenever `buildMetadataPath` is called out with a `V1` SecretStore.

Reply:
Hi Gustavo, this is true for the previous version of this PR when I removed the metadata validation blocks. 
Now, when we do support metadata in KV v1, I think that this logic should be supported as well.